UI redesign

scholvat opened the pull request:
Updated UI based off of Charlotte's design. Sorry for the long change list, but it's hard to compartmentalize a full UI redesign (I think 50 files changed in a PR is a new record for me).

For perspective, this is part 1 of a 2 part update. Part 2 will add widgets, as shown in Charlotte's Figma (https://www.figma.com/proto/UYSZRMkrGmkk0gJufFmk7eaj/Skybunk-april-11?node-id=1%3A2&scaling=scale-down). To make that look good requires a lot more code changes, and adding more than one widget. Hopefully this can be done by before start of term, but it might not. Because of this, Find a Don is still located under settings, and one of the new icons is not being used yet.

Changes:
-Changed color scheme to red and white
-More heavily used global stylesheet
-Changed navigation to make it more intuitive and compact
-Updated lots of icons
-Changed home footer tab to be icons instead of text (in preparation for widgets, which will be an upcoming PR)
-Added profile pictures to memberlist (this is reliant on https://github.com/CGUC/skybunk-server/pull/86 to have good performance)
-Refactored NotificationList to a FlatList for performance, pull to refresh (#162 ) and layout of "Mark Notifications as Read" button
-Lots of minor UI tweaks, as specified from Charlotte's redesign

#182

scholvat commented:
Screenshots can be found here: https://photos.app.goo.gl/rcu4Q88u5NDA5TnW9

CharlotteBaker316 commented:
Looks awesome to me! Can we deploy by Sunday?

picklechips commented:
I'm gonna test this out quick today just to double check everything still works, but screenshots look good

picklechips commented on components/ChannelList/ChannelList.jsx:
```diff
@@ -96,7 +96,7 @@ export default class ChannelList extends React.Component {
         }
         else if (subIndex !== -1) {
-          icon = require('../../assets/Bell-ON.png');
+          icon = require('../../assets/bell-ON.png');
```
`bell-ON.png` doesn't exist, the capitalisation doesn't match the actual asset.

scholvat replied:
I tried to make the filenames consistent, but forgot that windows has case insensitive filesystem. Should be fixed now.

picklechips commented:
App crashes for me when trying to view my notifications:

![image](https://user-images.githubusercontent.com/6520243/57152733-44eca600-6da2-11e9-9c89-fd4223792e76.png)

picklechips commented:
The background image on the login page won't load for me... I'm not sure if this has to do with this PR or not, but it doesn't happen on the current production app anyways.

![image](https://user-images.githubusercontent.com/6520243/57153023-3652be80-6da3-11e9-8681-47774efefb6e.png)

picklechips commented:
The skybunk logo isn't centred for me on the list of people

![image](https://user-images.githubusercontent.com/6520243/57153050-48346180-6da3-11e9-8da8-d05bb624bd29.png)

picklechips commented:
Aside from the few little things mentioned above, looks _really_ good. Love the new design

scholvat commented:
In terms of the loading screen image, I believe that is just a dev thing since a lot of the images are slow to load on expo. It works on production for me, and if it is an issue with other phones then I believe it is an issue separate from this UI update.

The member list header is an Android issue, and didn't catch it when doing my cross platform testing... I've changed it to text there since it will be centred and it makes more sense anyways.

That notification error is related to notifications of deleted posts. Should be fixed on the server side, but I've updated the error checking Neil put it to fix that.
